### PR TITLE
Bump lima to 0.8.3 and make static hostnames available inside containers

### LINKED ISSUE #1316: Lima: resolve host.docker.internal
### Preflight Checklist

- [X] I have searched the [issue tracker](https://github.com/rancher-sandbox/rancher-desktop/issues) for a feature request that matches the one I want to file, without success.

### Problem Description

We should be able to resolve `host.docker.internal` (and `host.minikube.internal`) on lima / macos.

### Proposed Solution

Something to do with the lima guest agent?

### Additional Information

See [docker documentation](https://docs.docker.com/desktop/mac/networking/#i-want-to-connect-from-a-container-to-a-service-on-the-host).

### LINKED ISSUE #1439: sudo: unable to resolve host lima-rancher-desktop: Name or service not known
https://github.com/rancher-sandbox/rancher-desktop/blob/1c4c40c3610ce0f448057e4d16330448ae33b967/src/assets/lima-config.yaml#L41-L46

`hostname` sets the `hostname`, but because it isn't added to `/etc/hosts`, programs, such as `sudo` will scream.

### PR DESCRIPTION
**Pull request by jandubois**
Tested manually (ping statistics removed for clarity):

```console
$ rdlima ping -c 1 lima-rancher-desktop
PING lima-rancher-desktop (192.168.5.15): 56 data bytes
64 bytes from 192.168.5.15: seq=0 ttl=42 time=0.157 ms

$ rdlima ping -c 1 host.docker.internal
PING host.docker.internal (192.168.5.2): 56 data bytes
64 bytes from 192.168.5.2: seq=0 ttl=42 time=0.349 ms

$ nerdctl run --rm alpine ping -c 1 lima-rancher-desktop
PING lima-rancher-desktop (192.168.5.15): 56 data bytes
64 bytes from 192.168.5.15: seq=0 ttl=64 time=0.043 ms

$ nerdctl run --rm alpine ping -c 1 host.docker.internal
PING host.docker.internal (192.168.5.2): 56 data bytes
64 bytes from 192.168.5.2: seq=0 ttl=254 time=0.340 ms
```

Fixes #1316
Fixes #1439

### CODE REVIEW

File: src/k8s-engine/lima.ts [hunk 1]
```diff
@@ -541,8 +544,17 @@ export default class LimaBackend extends events.EventEmitter implements K8s.Kube
...
       ],
-      ssh: { localPort: await this.sshPort },
-      k3s: { version: desiredVersion.version },
+      ssh:          { localPort: await this.sshPort },
+      k3s:          { version: desiredVersion.version },
+      hostResolver: {
+        hosts: {
+          // As far as lima is concerned, the instance name is 'lima-0'.
+          // We change the hostname in a provisioning script.
+          'lima-rancher-desktop':          'lima-0',
```
Comment (mook-as) [inline src/k8s-engine/lima.ts:553]: Presumably that should be `lima-${ MACHINE_NAME }`, but 🤷

**Review by mook-as — APPROVED** [review]

**Comment by johnou** [comment]
@jandubois I assume this will be part of 1.1.0? do you have any dates planned?

**Comment by jandubois** [comment]
> @jandubois I assume this will be part of 1.1.0? do you have any dates planned?

Yes, it will be part of the next release later this week.

**Comment by johnou** [comment]
@jandubois for what its worth, I tried working around this in docker compose on dockerd with the following

```
    extra_hosts:
      - "host.docker.internal:host-gateway"
```

but the container routed host.docker.internal to 172.17.0.1 instead of expected qemu gateway address of 192.168.5.2, is this expected and will your fix address this? If I have time tomorrow I might try building rancher from source.

**Comment by jandubois** [comment]
> but the container routed host.docker.internal to 172.17.0.1

I think `host-gateway` will be the gateway to the container host, which is the VM, and not the VM host. So if you specify it as an extra host, I expect it will take precedence over the address provided by DNS. You will have to remove it from `extra_hosts` to get it working.

BTW, you don't need to build from source, you could also just download a build from CI. You must be logged into Github (otherwise you won't get the asset links), and then look at the Github Actions at https://github.com/rancher-sandbox/rancher-desktop/actions.

Choose the latest "package" action of a merge to `main`, e.g. https://github.com/rancher-sandbox/rancher-desktop/actions/runs/1901120900. You may need to scroll down a bit to see all the assets produced by that action.

If you download the macOS bits, you will need to clear the quarantine xattr before you can run them because CI builds have not been notarized and signed yet.

**Comment by johnou** [comment]
@jandubois I can confirm my use case is resolved in build 362 (g470f7cb) thanks a lot!

**Comment by mfriess2** [comment]
We are trying to use a same docker-compose file for Linux and Mac with Rancher-Desktop.
Unfortunately, for Linux it is mandatory to have the extra-hosts entry in the compose file:
```
    extra_hosts:
      - "host.docker.internal:host-gateway"
```
but as @johnou already mentioned, with Rancher-Desktop (or Lima !?) this will resolve to "172.17.0.1" instead of "192.168.5.2" when the containers are running.

Is there any solution how to get this working cross-platform for Linux and MacOS-Rancher compose files?

**Comment by johnou** [comment]
@mfriess2 we use https://github.com/qoomon/docker-host in our project which works with Windows, Linux and OSX.

eg.

```
version: "3.4"

services:
  docker-host:
    image: qoomon/docker-host
    cap_add: [ 'NET_ADMIN', 'NET_RAW' ]
    restart: on-failure
    networks:
      fuse:
        aliases:
          - localhost.acme.com
```

then from other containers localhost.acme.com will point to the VM host.

**Pull request merged**